### PR TITLE
interfaces: add network-manager-observe interface

### PR DESCRIPTION
**Pull request by jdstrand**
NetworkManager offers a very large API that is wholly exposed via the
network-manager interface. Some applications (eg, QtNetworkBearer) use
NetworkManager to check for connectivity and therefore do not require
access to the entire API. Implement the network-manager-observe
interface that exposes the Get* and List methods required for verifying
connectivity. Since this grants access to things like MAC addresses and
previous networks, treat this like network-observe and require manual
connection. This interface does not grant access to NetworkManager's
GetSecrets() API.
    
On classic distro, this interface lets the application communicate with
the system NetworkManager using the specified APIs. On all snaps
systems, a providing snap interested in exposing this smaller API is
expected to slot this interface in addition to the fuller
network-manager interface, which contains all the permanent slot policy
needed to run as NetworkManager (ie, this interface is currently not
standalone as there are no known applications that only implement this
reduced NM API).

References:
https://forum.snapcraft.io/t/qt-network-bearer-networkmanager-access-extend-network-observe/8399

### CODE REVIEW

File: interfaces/builtin/network_manager_observe.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+        - app
+        - core
+    deny-auto-connection: true
+    deny-connection:
+      on-classic: false
+`
+
+const networkManagerObserveSummary = `allows observing NetworkManager settings`
+
+const networkManagerObserveConnectedSlotAppArmor = `
```
Comment (bboozzoo) [inline interfaces/builtin/network_manager_observe.go]: NM sends a couple of signals that could be useful too: org.freedesktop.NetworkManger.PropertiesChanged and org.freedesktop.NetworkManager.StateChanged on /org/freedesktop/NetworkManger. Note that org.freedesktop.NetworkManger.PropertiesChanged  isn't exactly the same as freedeskto.DBus.Properties.PropertiesChanged, but while at it we may want to allow both.

There is also a couple of potentially useful signals in the Settings interface (/org/freedesktop/NetworkManager/Settings path)

Comment (bboozzoo) [reply interfaces/builtin/network_manager_observe.go]: Looked at `gio`, the `GNetworkMonitor` backend for NetworkManager uses `org.freedesktop.NetworkManger.PropertiesChanged` signal to trigger the refresh of a bunch of variables.

Comment (jdstrand) [reply interfaces/builtin/network_manager_observe.go]: Ok, I added the corresponding signal rules for allowed method rules. I'm told this is all that is required by QtNetworkBearer; we can iterate on the interface as needed (but the signals you mentioned should be covered).

Comment (jdstrand) [reply interfaces/builtin/network_manager_observe.go]: > NM sends a couple of signals that could be useful too: org.freedesktop.NetworkManger.PropertiesChanged and org.freedesktop.NetworkManager.StateChanged on /org/freedesktop/NetworkManger. Note that org.freedesktop.NetworkManger.PropertiesChanged isn't exactly the same as freedeskto.DBus.Properties.PropertiesChanged, but while at it we may want to allow both.
> 
> There is also a couple of potentially useful signals in the Settings interface (/org/freedesktop/NetworkManager/Settings path)

@bboozzoo - this is done.

**Pull request merged**